### PR TITLE
Support query hints

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

**Review by jdm-square — APPROVED** [review]
Love the test for this!

**Comment by jdm-square** [comment]
You will need to follow https://github.com/cashapp/misk/blob/c19582e42bc18f6effd20b50ba99993658b55fe0/CONTRIBUTING.md#breaking-changes as well.

**Review by swankjesse — APPROVED** [review]

**Review by frojasg — APPROVED** [review]
LGTM

File: misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt [hunk 1]
```diff
@@ -1155,4 +1156,48 @@ class ReflectionQueryFactoryTest {
...
+          .contains("use index (unq_name)")
+
+        assertThat(
+          queryFactory.newQuery<OperatorsMovieQuery>()
+            .queryHint("unq_name")
+            .list(session)
+        )
+        assertThat(logCollector.takeMessage(null, Level.DEBUG, Regex("/* select.*from movies")))
+          .contains("use index (unq_name)")
+      }
```
[the remark this answers is not in the document]
Comment (tirsen) [reply misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt:1188]: Yes I was also surprised by this, it seems Hibernate only supports `use index` hints. Which is pretty weird.

**Pull request merged**